src/gpu/nvidia/cudnn_convolution_impl.hpp

```diff
@@ -440,19 +460,37 @@ struct cudnn_convolution_impl_fwd_t : public cudnn_convolution_impl_base_t {
         // If the only post-op is fused then there is no need for temp dst
         if (conv_bias_eltwise && num_post_ops == 1) use_temp_dst_ = false;
 
-        if (data_types[y] == CUDNN_DATA_INT8 && use_temp_dst_) {
+        // We need to take into account if we are scaling
+        // the src. In which case we will need to compute
+        // the post-ops in f32 and then quantize using a
+        // 1.0f scaling factor for dst
+        if (data_types[y] == CUDNN_DATA_INT8 && use_temp_dst_
+                && !(do_dst_scaling || do_src_scaling)) {
             data_types[y] = CUDNN_DATA_FLOAT;
             need_reorder = true;
             CHECK(create_and_set_tensor_descriptor_ex(&reorder_dst_desc,
                     formats[y], reorder_type, ndims[y], dims[y]));
         }
 
+        // If dst needs to be scaled and dst datatype is s8
+        if ((do_dst_scaling || do_src_scaling)
```

Comment:
Thanks for the fixes, it is getting better.

There is also a weights scale. With the current implementation the following case fails:
```
$ ./build/tests/benchdnn/benchdnn --conv --engine=gpu --skip-impl=ref --dir=FWD_I --dt=s8:s8:s8 --attr-scales=wei:common:2 --attr-post-ops=sum mb1ic512iw121oc512ow122kw6pw3nconv1d:21
[   4][DST][0:0:4] exp_f32:        -156 exp:        -128 got:        -124 diff:       4 rdiff: 0.03125
[   8][DST][0:0:8] exp_f32:         141 exp:         127 got:         124 diff:       3 rdiff:0.023622
[  21][DST][0:0:21] exp_f32:         257 exp:         127 got:         124 diff:       3 rdiff:0.023622
[  36][DST][0:0:36] exp_f32:        -183 exp:        -128 got:        -127 diff:       1 rdiff:0.0078125
[  46][DST][0:0:46] exp_f32:         258 exp:         127 got:         125 diff:       2 rdiff:0.015748
[  60][DST][0:0:60] exp_f32:         137 exp:         127 got:         124 diff:       3 rdiff:0.023622
[  69][DST][0:0:69] exp_f32:        -161 exp:        -128 got:        -125 diff:       3 rdiff:0.0234375
[  74][DST][0:0:74] exp_f32:         140 exp:         127 got:         123 diff:       4 rdiff:0.0314961
[  88][DST][0:0:88] exp_f32:         129 exp:         127 got:         124 diff:       3 rdiff:0.023622
[  92][DST][0:0:92] exp_f32:         250 exp:         127 got:         125 diff:       2 rdiff:0.015748
[COMPARE_STATS][DST]: trh=0 max_diff:       4 max_rdiff:0.0314961
0:FAILED (errors:5371 total:62464) __REPRO: --conv --engine=gpu --skip-impl=ref --dir=FWD_I --dt=s8:s8:s8 --attr-scales=wei:common:2 --attr-post-ops=sum mb1ic512iw121oc512ow122kw6pw3nconv1d:21
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total: 1.68s; fill: 0.02s (1%); compute_ref: 0.02s (1%); compare: 0.01s (0%);

```

Reply:
Thanks for the feedback again @igorsafo.
I added the code to take care of weight scales. I run the example and now it is working.
Please let me know about any new potential changes to apply.

Reply:
@kala855  thanks, now it works on my side. There is another issue related to the sum post op. It fails with or without scales, but only with s8 dst type:
```
$ ./build/tests/benchdnn/benchdnn --conv --engine=gpu --skip-impl=ref --dir=FWD_D --dt=s8:s8:s8  --attr-post-ops=sum:2 g1ic32ih149oc32oh147kh3ph0n"googlenet_v3:conv_1_1_conv2d"
[ 140][DST][0:0:0:140] exp_f32:         126 exp:         126 got:         121 diff:       5 rdiff:0.0396825
[ 186][DST][0:0:1:39] exp_f32:        -132 exp:        -128 got:        -120 diff:       8 rdiff:  0.0625
[ 363][DST][0:0:2:69] exp_f32:        -130 exp:        -128 got:        -126 diff:       2 rdiff:0.015625
[ 574][DST][0:0:3:133] exp_f32:        -154 exp:        -128 got:        -124 diff:       4 rdiff: 0.03125
[ 607][DST][0:0:4:19] exp_f32:        -146 exp:        -128 got:        -124 diff:       4 rdiff: 0.03125
[ 894][DST][0:0:6:12] exp_f32:        -126 exp:        -126 got:        -122 diff:       4 rdiff:0.031746
[1188][DST][0:0:8:12] exp_f32:         148 exp:         127 got:         121 diff:       6 rdiff:0.0472441
[1551][DST][0:0:10:81] exp_f32:         142 exp:         127 got:         123 diff:       4 rdiff:0.0314961
[1680][DST][0:0:11:63] exp_f32:        -154 exp:        -128 got:        -120 diff:       8 rdiff:  0.0625
[1716][DST][0:0:11:99] exp_f32:        -134 exp:        -128 got:        -126 diff:       2 rdiff:0.015625
[COMPARE_STATS][DST]: trh=0 max_diff:       8 max_rdiff:0.0629921
0:FAILED (errors:7570 total:1382976) __REPRO: --conv --engine=gpu --skip-impl=ref --dir=FWD_D --dt=s8:s8:s8 --attr-post-ops=sum:2 g1ic32ih149oc32oh147kh3ph0ngooglenet_v3:conv_1_1_conv2d
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total: 1.69s; fill: 0.04s (2%); compute_ref: 0.03s (2%); compare: 0.05s (3%);
$ ./build/tests/benchdnn/benchdnn --conv --engine=gpu --skip-impl=ref --dir=FWD_D --dt=s8:s8:f32  --attr-post-ops=sum:2 g1ic32ih149oc32oh147kh3ph0n"googlenet_v3:conv_1_1_conv2d"
0:PASSED __REPRO: --conv --engine=gpu --skip-impl=ref --dir=FWD_D --dt=s8:s8:f32 --attr-post-ops=sum:2 g1ic32ih149oc32oh147kh3ph0ngooglenet_v3:conv_1_1_conv2d
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:0 listed:0
total: 1.64s; fill: 0.03s (2%); compute_ref: 0.05s (3%); compare: 0.01s (1%);
```

I found the issue by running `./build/tests/benchdnn/benchdnn --conv --engine=gpu --batch=tests/benchdnn/inputs/conv/test_conv_int8`

Reply:
@igorsafo Thanks, I was checking this issue but it seems like it is not related to the scaling parameter issues with int8 as reported in issue #1749. I could open and start solving a new issue to deal with this sum post-op problem. Let me know if it is OK. Thank you.

Reply:
@kala855 , Agree, I was able to reproduce the issue prior to the changes in this PR.
Here is a separate issue: https://github.com/oneapi-src/oneDNN/issues/1870